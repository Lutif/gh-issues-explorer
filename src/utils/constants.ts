export const PAGE_SIZE = 10;
export const REACT_APP_GITHUB_BASE_URL = "https://api.github.com/graphql";
