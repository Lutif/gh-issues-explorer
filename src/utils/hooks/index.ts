export * from "./useSearchIssue";
export * from "./useFetchIssueDetails";
