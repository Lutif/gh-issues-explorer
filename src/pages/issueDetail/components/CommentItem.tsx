import moment from "moment";
import { Comment } from "../../../utils";
import { CommentItemWrapper } from "./commentItem.styles";

interface CommentItemProps {
  comment: Comment;
}

export const CommentItem: React.FC<CommentItemProps> = ({ comment }) => {
  const { bodyHTML, author: { login } = {}, createdAt } = comment.node;
  const subText = `Commented ${moment(createdAt).fromNow()} by ${login}`;
  return (
    <CommentItemWrapper>
      <div
        className="content"
        dangerouslySetInnerHTML={{ __html: bodyHTML as string }}
      />
      <div className="sub-text">{subText}</div>
    </CommentItemWrapper>
  );
};
