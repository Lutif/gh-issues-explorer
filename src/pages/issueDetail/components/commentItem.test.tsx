import { render, screen } from "@testing-library/react";
import moment from "moment";
import { Comment } from "../../../utils";
import { CommentItem } from "./CommentItem";
describe("<CommentItem /> tests", () => {
  const commentBody = "test comment";

  const testComment: Comment = {
    node: {
      bodyHTML: `<p>${commentBody}</p>`,
      createdAt: new Date(),
      author: {
        login: "lutif",
        avatarUrl: "test",
        url: "test",
      },
    },
  };
  const subText = `Commented ${moment(
    testComment.node.createdAt
  ).fromNow()} by ${testComment.node.author.login}`;
  it("renders the CommentItem component", () => {
    render(<CommentItem comment={testComment} />);
    const element = screen.getByText(commentBody);
    expect(element).toBeInTheDocument();
  });

  it("test CommentItem snapshot", () => {
    const { asFragment } = render(<CommentItem comment={testComment} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("test commentItem author name in document", () => {
    render(<CommentItem comment={testComment} />);
    const element = screen.getByText(subText);
    expect(element).toBeInTheDocument();
  });
});
