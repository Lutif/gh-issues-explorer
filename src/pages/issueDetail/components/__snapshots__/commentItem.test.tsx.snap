// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentItem /> tests test CommentItem snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jbcHnV"
  >
    <div
      class="content"
    >
      <p>
        test comment
      </p>
    </div>
    <div
      class="sub-text"
    >
      Commented a few seconds ago by lutif
    </div>
  </div>
</DocumentFragment>
`;
